hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/CarboniteWorkerActor.scala

```diff
@@ -1,47 +1,94 @@
+class CarboniteWorkerActor(carboniterConfig: HybridCarboniteConfig,
+                           serviceRegistryActor: ActorRef,
+                           ioActor: ActorRef) extends Actor with ActorLogging with GracefulShutdownHelper {
 
   implicit val ec: ExecutionContext = context.dispatcher
 
-  scheduleNextCarboniting()
+  val carboniteFreezerActor = context.actorOf(CarbonitingMetadataFreezerActor.props(carboniterConfig, self, serviceRegistryActor, ioActor))
+
+  val findWorkflowToCarboniteQuery: Seq[(String, String)] = Seq(
+    IncludeSubworkflows.name -> "false",
+    Status.name -> WorkflowSucceeded.toString,
+    Status.name -> WorkflowFailed.toString,
+    Status.name -> WorkflowAborted.toString,
+    MetadataArchiveStatus.name -> Unarchived.toString,
+    Page.name -> "1",
+    PageSize.name -> "1"
+  )
+
+  val backOff = SimpleExponentialBackoff(
+    initialInterval = 5.seconds,
+    maxInterval = 5.minutes,
+    multiplier = 1.1,
+    randomizationFactor = 0.0
+  )
+
+  scheduleNextCarboniting(backOff.backoffMillis)
+
 
   override def receive: Receive = {
-    case DoCarboniting => doCarboniting()
-    case ShutdownCommand => stopGracefully()
-    case oops => log.error(s"Programmer Error! The CarboniteWorkerActor is not to be talked to! ($sender sent $oops})")
+    case s: WorkflowQuerySuccess => {
+      if (s.response.results.nonEmpty) carboniteWorkflow(s.response.results.head.id)
+      else scheduleNextCarboniting(backOff.backoffMillis)
+    }
+    case f: WorkflowQueryFailure => {
+      log.error(s"Something went wrong while querying to find workflow to carbonite. It will retry in sometime. Error: ${f.reason}")
+      scheduleNextCarboniting(backOff.backoffMillis)
+    }
+    case c: CarboniteWorkflowComplete => {
+      // after completion of carbonite process reset the backoff
+      log.info(s"Carbonation complete for workflow ${c.workflowId}")
+      backOff.googleBackoff.reset()
+      scheduleNextCarboniting(backOff.backoffMillis)
+    }
+    case ShutdownCommand => waitForActorsAndShutdown(NonEmptyList.of(carboniteFreezerActor))
+    case other => log.error(s"Programmer Error! The CarboniteWorkerActor received unexpected message! ($sender sent $other})")
   }
 
-  def doCarboniting(): Unit = {
-    // TODO: [CARBONITE] When carboniting actually does something, we probably metrics here rather than a tick log...
-    log.info("Carbonite Worker Tick...")
 
-    scheduleNextCarboniting()
+  def scheduleNextCarboniting(interval: Long): Unit = {
+    val duration = Duration(interval, MILLISECONDS)
+    context.system.scheduler.scheduleOnce(duration)(findWorkflowToCarbonite())
+    ()
   }
 
-  def scheduleNextCarboniting(): Unit = {
-    context.system.scheduler.scheduleOnce(carboniteInterval) {
-      self ! DoCarboniting
-    }
-    ()
+
+  def findWorkflowToCarbonite(): Unit = {
+    // TODO: [CARBONITE] When carboniting actually does something, we probably metrics here rather than a tick log...
```

Comment:
```suggestion
    // TODO: [CARBONITE] When carboniting actually does something, we should probably publish metrics here rather than log a tick...
```